clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ConsumptionTask.java

```diff
@@ -117,14 +116,17 @@ public void run() {
          * and thus be capable of operating on a non-threadsafe consumer.
          */
         polledPubSubMessages = pollFunction.get();
-        polledPubSubMessagesCount = polledPubSubMessages.count();
+        polledPubSubMessagesCount = polledPubSubMessages.values().stream().mapToInt(List::size).sum();
```

Comment:
Instead of having a dedicated loop for this here, can we increment a counter in the first for loop below (the one going over `polledPubSubMessages.entrySet()`)? We can record the stat after that.

Reply:
Good point. I will do that.